### PR TITLE
New Navigation UX

### PR DESCRIPTION
**Pull request by sagerb**
Changes implement a new UX to the user guide, allowing explicit navigation to the user guide, the setup instructions for Tableau, the OpenAPI page as well as some help.

#### Without ERRORS

Default page from Connect (also accessible via HOME icon click)
2021-08-02 at 3 11 PM

Mouse over on Home icon (or clicking on it)
2021-08-02 at 3 11 PM

Mouse over on Gears and click brings up the Tableau setup instructions
2021-08-02 at 3 11 PM

Mouse over on Info circle (I) and click brings up the OpenAPI page
2021-08-02 at 2 55 PM

Mouse over on Question mark circle (?) and click brings up the help which we'll show some of the help from the vignettes.
2021-08-02 at 3 12 PM

#### With ERRORS

Default page from Connect (also accessible via HOME icon click)
2021-08-02 at 3 12 PM

Mouse over on Home icon (or clicking on it)
2021-08-02 at 3 12 PM

Mouse over on Gears and click brings up the Tableau setup instructions
2021-08-02 at 3 13 PM

Mouse over on Info circle (I) and click brings up the OpenAPI page
2021-08-02 at 2 58 PM

Mouse over on Question mark circle (?) and click brings up the help which we'll show some of the help from the vignettes.
2021-08-02 at 3 13 PM

### CODE REVIEW

File: R/user_guide.R [hunk 1]
```diff
@@ -40,66 +40,68 @@ create_user_guide <- function(pr) {
...
     )
-  } else {
-    apiSpec <- pr$getApiSpec()
 
-    title <- apiSpec$info$title
-    version <- apiSpec$info$version
+    body_content <- htmltools::tagList(
+      tags$h3(
+        class="warning",
+        "Warning: The following item(s) need to be resolved before your API will be accessible from Tableau!"
```
Comment (toph-allen) [inline R/user_guide.R:63]:
```suggestion
        "Warning: The following item(s) need to be resolved before your API will be accessible from Tableau."
```

File: R/user_guide.R [hunk 2]
```diff
@@ -40,66 +40,68 @@ create_user_guide <- function(pr) {
...
+          class="padded-flat-top",
+          htmltools::HTML(desc)
         )
       )
     )
+    
+    body_content <- htmltools::tagList(
+      tags$h3(
+        class="subtitle",
+        "Use your analytics extension(s) from Tableau"
```
Comment (toph-allen) [inline R/user_guide.R:92]:
```suggestion
        "Use this extension in Tableau"
```
- I'm torn between "your extension" and "this extension". Leaning towards "this".
- Only need the singular because the generated code is only for this single extension.
- I still don't think this gets across "Auto-generated documentation and usage for _this_ extension. Not sure how we would though. But since this is the first page that shows, I think it'll get the message across.

File: R/user_guide.R [hunk 3]
```diff
@@ -281,60 +285,229 @@ render_setup_instructions <- function(path, pr) {
...
+        class="padded-flat-top",
+        htmltools::HTML(desc)
       )
+    )
+  )
+
+  body_content <- htmltools::tagList(
+    tags$h3(
+      class="subtitle",
+      "Configure Tableau to access your extension"
```
Comment (toph-allen) [inline R/user_guide.R:315]:
```suggestion
      "Configure Tableau to access this extension"
```
The more I think about it, the more "this" implies that the page might be tailored to _this_ one.

File: R/user_guide.R [hunk 4]
```diff
@@ -281,60 +285,229 @@ render_setup_instructions <- function(path, pr) {
...
+  as.character(htmltools::htmlTemplate(
+    system.file("template/index.html", package = "plumbertableau", mustWork = TRUE),
+    title_desc = title_desc,
+    body_content = body_content
+  ))
+}
+
+help_text <- function() {
+"
+The plumbertableau package makes it simple for R developers to build [Tableau Analytics Extensions](https://tableau.github.io/analytics-extensions-api/) by lightly annotating Plumber APIs and deploy those extensions to [RStudio Connect](https://www.rstudio.com/products/connect/). It also makes it easy for Tableau developers to use those extensions, by automatically generating instructions and code examples tailored for Tableau.
```
Comment (toph-allen) [inline R/user_guide.R:482]: This text is all from the old version of the vignettes, before the rewrite that landed on Friday. Should change it to reflect those. I'll take a look in a few minutes to see if there's any section I think works.

File: inst/template/index.html [hunk 5]
```diff
@@ -6,12 +6,93 @@
...
+                    aria-hidden="true" 
+                    role="img" 
+                    viewBox="0 0 512 512" 
+                    style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"
+                    >
+                      <path d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"></path>
+                  </svg>
+                </div>
+                <span class="nav-text">
+                  View your extension's Open API documentation
```
Comment (toph-allen) [inline inst/template/index.html:64]:
```suggestion
                  View this extension's OpenAPI documentation
```

File: inst/template/index.html [hunk 6]
```diff
@@ -6,12 +6,93 @@
...
+                    aria-hidden="true" 
+                    role="img" 
+                    viewBox="0 0 512 512" 
+                    style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"
+                  >
+                    <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"></path>
+                  </svg>
+                </div>
+                <span class="nav-text">
+                  Help with plumbertableau and RStudio Connect
```
Comment (toph-allen) [inline inst/template/index.html:82]:
```suggestion
                  Read up on plumbertableau, Tableau, and RStudio Connect
```
Changed this heading to match syntactically with the others. "Read up on" is a little informal, but I like the sound of it. Better than other things I considered, like "Get help with" (which suggests you might be able to ask someone questions) or "Read more about" (which sounds generic and marketing-y).

**Review by toph-allen — CHANGES_REQUESTED** [review]
Looks good, except for that we need to update the stuff on the (?) page to match with the rewritten vignettes.

File: R/user_guide.R (hunk 4 above)
Comment (toph-allen) [reply R/user_guide.R:482]: I suggest we pull text from this vignette: https://rstudio.github.io/plumbertableau/articles/introduction.html.

File: R/user_guide.R (hunk 3 above)
Comment (sagerb) [reply R/user_guide.R]: I can't commit your change, but I will make the change myself that you recommended.

File: R/openapi.R [hunk 7]
```diff
@@ -35,13 +40,16 @@ tableau_openapi <- function(pr) {
...
+        "### Description\n",
         spec$info$description,
+        "\n",
+        "#### Use the following links to setup and use your Tableau Analytics Extension.",
+        "\n",
+        "* [Use your analytics extension from Tableau](../)",
+        "\n",
+        "* [Configure Tableau to use your analytics extension](../setup)",
+        "\n",
+        "* [Read up on plumbertableau, Tableau, and RStudio Connect](../help)",
```
Comment (blairj09) [inline R/openapi.R:52]:
```suggestion
        "* [Additional details about plumbertableau, Tableau, and RStudio Connect](../help)",
```

File: inst/template/index.html [hunk 8]
```diff
@@ -6,12 +6,93 @@
...
+                    aria-hidden="true" 
+                    role="img" 
+                    viewBox="0 0 512 512" 
+                    style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"
+                  >
+                    <path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"></path>
+                  </svg>
+                </div>
+                <span class="nav-text">
+                  Read up on plumbertableau, Tableau, and RStudio Connect
```
Comment (blairj09) [inline inst/template/index.html:82]:
```suggestion
                  Additional details about plumbertableau, Tableau, and RStudio Connect
```

File: R/user_guide.R [hunk 9]
```diff
@@ -281,60 +272,209 @@ render_setup_instructions <- function(path, pr) {
...
     system.file("template/index.html", package = "plumbertableau", mustWork = TRUE),
-    content = ui
+    title_desc = title_desc,
+    body_content = body_content
   ))
 }
 
-strip_md_links <- function(text) {
-  stringi::stri_replace_all(text, regex = " +\\*.*", "")
+help_text <- function() {
```
Comment (blairj09) [inline R/user_guide.R:464]: Agree with @toph-allen that we'll want this to come directly from the vignette if possible, but that can be explored in a separate PR

**Review by blairj09 — APPROVED** [review]
A few minor suggestions, but LGTM!

File: inst/template/index.html (hunk 8 above)
Comment (sagerb) [reply inst/template/index.html]: updated
